### PR TITLE
[Observability] Fix --follow lost connection when it is used for > 30 seconds

### LINKED ISSUE #25721: [Core] [State Observability] `--follow` lost connection when it is used for > 30 seconds.
### What happened + What you expected to happen

```
Traceback (most recent call last):
  File "/Users/sangbincho/anaconda3/envs/core/bin/ray", line 33, in <module>
    sys.exit(load_entry_point('ray', 'console_scripts', 'ray')())
  File "/Users/sangbincho/work/ray/python/ray/scripts/scripts.py", line 2496, in main
    return cli()
  File "/Users/sangbincho/anaconda3/envs/core/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/Users/sangbincho/anaconda3/envs/core/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/Users/sangbincho/anaconda3/envs/core/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/sangbincho/anaconda3/envs/core/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/sangbincho/anaconda3/envs/core/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/Users/sangbincho/work/ray/python/ray/scripts/scripts.py", line 2079, in logs
    for chunk in get_log(
  File "/Users/sangbincho/work/ray/python/ray/experimental/state/api.py", line 264, in get_log
    raise RayStateApiException(error_msg)
ray.experimental.state.exception.RayStateApiException: 0Closing HTTP stream due to internal server error.
<AioRpcError of RPC that terminated with:
	status = StatusCode.DEADLINE_EXCEEDED
	details = "Deadline Exceeded"
	debug_error_string = "{"created":"@1655126573.655659000","description":"Error received from peer ipv4:127.0.0.1:58511","file":"src/core/lib/surface/call.cc","file_line":1075,"grpc_message":"Deadline Exceeded","grpc_status":4}"
```

When I ran ray logs dashboard.log --follow, I got this error after 30 seconds. It is probably that we have a timeout when --follow is used which should be avoided (if we follow, we'd just like to keepalive rather than timeout)

### Versions / Dependencies

Master

### Reproduction script

Run `ray logs dashboard.log --follow` for a long time

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR fixes the issue where --follow lost connection when it is used for > 30 seconds because the gRPC timeout is configured to be 30 seconds, and we don't reset it when --follow is set.

This fixes the issue by setting timeout=None when keepalive==True

## Related issue number

Closes https://github.com/ray-project/ray/issues/25721

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/log/log_manager.py [hunk 1]
```diff
@@ -88,7 +88,10 @@ async def stream_logs(
             keep_alive=keep_alive,
             lines=options.lines,
             interval=options.interval,
-            timeout=options.timeout,
+            # If we keepalive logs connection, we shouldn't have timeout
+            # otherwise the stream will be terminated forcefully
+            # after the deadline is expired.
+            timeout=options.timeout if not keep_alive else None,
```
Comment (scv119) [inline dashboard/modules/log/log_manager.py:94]: i think there are multiple timeout: connection time/receive timeout. Does the client allow us to specify connection timeout? For streaming we probably want to handle connection timeout?

Comment (rickyyx) [reply dashboard/modules/log/log_manager.py:94]: Right, I think we could specify (connect timeout, read timeout) as a pair. https://requests.readthedocs.io/en/latest/user/advanced/#timeouts

Comment (rkooo567) [reply dashboard/modules/log/log_manager.py:94]: This is using gRPC, not requests. I think connection timeout should be automatically handled by gRPC

**Comment by rkooo567** [comment]
Ping for the follow up request or approval!

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/disneypixar/">Disney Pixar</a> on <a href="https://giphy.com/gifs/disneypixar-cute-disney-cartoon-WFyvwq3xoTTMs">GIPHY</a></div>

**Review by scv119 — APPROVED** [review]

**Pull request merged**